rcl/src/rcl/init.c

```diff
@@ -184,13 +201,22 @@ rcl_shutdown(rcl_context_t * context)
     RCL_SET_ERROR_MSG(rmw_get_error_string().str);
     return rcl_convert_rmw_ret_to_rcl_ret(rmw_ret);
   }
-
-  rcl_ret_t rcl_ret = rcl_logging_fini();
-  RCUTILS_LOG_ERROR_EXPRESSION_NAMED(
-    RCL_RET_OK != rcl_ret, ROS_PACKAGE_NAME,
-    "Failed to fini logging, rcl_ret_t: %i, rcl_error_str: %s", rcl_ret,
-    rcl_get_error_string().str);
-  rcl_reset_error();
+  size_t previous_logging_ref_count =
+    rcutils_atomic_fetch_add_uint64_t(&g_logging_ref_count, -1);
+
+  if (1u == previous_logging_ref_count) {
```

Comment:
@ivanpauno it'd be nice to double check if `rcutils` atomics include memory barriers as well. I've mostly assumed so, but just in case.

Reply:
Yes: 
- no windows: https://github.com/ros2/rcutils/blob/d7bd0575d8eda3e2b19757350cb9169ffbbbcae4/include/rcutils/stdatomic_helper/gcc/stdatomic.h#L368,
- windows:
https://github.com/ros2/rcutils/blob/d7bd0575d8eda3e2b19757350cb9169ffbbbcae4/include/rcutils/stdatomic_helper/win32/stdatomic.h#L270
https://docs.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedexchangeadd64